### ISSUE TITLE
Isolate loader to start on click

### ISSUE THREAD

**Issue by starknetdev** [BibliothecaDAO/loot-survivor#97]
Experiencing loader delays between clicking a button for a task and waiting for the tx to start. With the intro to Cartridge and session keys the loader should trigger on click rather than tx submission.

Current:
```
  await handleSubmitCalls(writeAsync).then((tx: any) => {
    if (tx) {
      startLoading(
        "Explore",
        tx.transaction_hash,
        "Exploring",
        "discoveryByTxHashQuery",
        adventurer?.id
      );
      addTransaction({
        hash: tx.transaction_hash,
        metadata: {
          method: `Explore with ${adventurer?.name}`,
        },
      });
    }
 ```
 
 Proposed:
 ```
 startLoading(...) //NEW CODE
 await handleSubmitCalls(writeAsync).then((tx: any) => {
  if (tx) {
    setLoadTxHash(tx.transaction_hash) //NEW CODE
    addTransaction({
      hash: tx.transaction_hash,
      metadata: {
        method: `Explore with ${adventurer?.name}`,
      },
    });
  }
 ```

**Comment by ponderingdemocritus** [comment]
perfect!

**Comment by starknetdev** [comment]
Fixed in #107
